src/components/TextInputWithAssetSelector/TextInputWithAssetSelector.js

```diff
@@ -0,0 +1,102 @@
+// @flow
+/*
+    Pillar Wallet: the personal data locker
+    Copyright (C) 2019 Stiftung Pillar Project
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+import * as React from 'react';
+import TextInput from 'components/TextInput';
+import Input from 'components/Input';
+
+import type { Option } from 'models/Selector';
+
+import ExchangeInputHeader from './TextInputWithAssetSelectorHeader';
+
+type Props = {
+  onBlur?: () => void,
+  errorMessage?: string,
+  asset: Option,
+  onAssetPress: () => void,
+  labelText?: string,
+  onLabelPress?: () => void,
+  onChange?: (val: string) => void,
+  value: ?string,
+  leftSideText: string,
+  onLeftSideTextPress: () => void,
+  leftSideSymbol: string,
+  rightPlaceholder: string,
+  disabled?: boolean,
+  getInputRef?: (Input) => void
+}
+
+export default class ExchangeTextInput extends React.PureComponent<Props> {
```

Comment:
Can we make it as function component? Not blocking as it's by preference, but makes things cleaner.